src/traces/scatter/fillcolor_defaults.js

```diff
@@ -2,6 +2,17 @@
 
 var Color = require('../../components/color');
 var isArrayOrTypedArray = require('../../lib').isArrayOrTypedArray;
+var assert = require('assert');
+
+function averageColors(colorscale) {
+    assert(colorscale.length >= 2);
```

Comment:
Let's not use `assert` in our code which is now in our dev-dependencies.
Instead we should move this kind of input validation to the supply defaults step and set the `fillgradient.colorscale` to the `dflt` value in case of bad inputs.

Reply:
Actually it looks like this function is unused and never get called?

Reply:
`averageColors` is called in `fillColorDefaults` as a fallback option to set `fillcolor` (if it was not explicitely set by the user and the color is not inherited from markers, but before using the `defaultColor`). 

I have removed the assertion, it seems that colorscale inputs are already checked to have length >= 2.